src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -79,13 +79,17 @@ bool PDPServer2::init(
         return false;
     }
 
+    // Initialize server dedicated thread.
+    resource_event_thread_.init_thread();
+
     /*
         Given the fact that a participant is either a client or a server the
         discoveryServer_client_syncperiod parameter has a context defined meaning.
      */
     mp_sync = new DServerEvent2(this,
-                    TimeConv::Duration_t2MilliSecondsDouble(m_discovery.discovery_config.
-                    discoveryServer_client_syncperiod));
+                    TimeConv::Duration_t2MilliSecondsDouble(
+                            m_discovery.discovery_config.discoveryServer_client_syncperiod));
+
     awakeServerThread();
```

Comment:
I'd leave the first awaken for the first data received/added to the database, either by listeners or `PDPServer2::announceParticipantState()`